src/loader/SpineLoader.ts

```diff
@@ -1,83 +1,119 @@
+import { SkeletonData, TextureAtlas } from "@esotericsoftware/spine-core";
 import {
   AssetPromise,
+  BufferReader,
   Loader,
   LoadItem,
   resourceLoader,
   ResourceManager,
-  Texture2D,
+  Texture2D
 } from "@galacean/engine";
-import { createSkeletonData, createTextureAtlas, loadTextureAtlas, loadTexturesByPath } from "./LoaderUtils";
+import {
+  createSkeletonData,
+  createTextureAtlasWithOriginAsset,
+  loadTextureAtlas,
+  loadTexturesByPaths
+} from "./LoaderUtils";
 import { SpineResource } from "./SpineResource";
-import { BufferReader } from "../util/BufferReader";
-import { TextureAtlas } from "@esotericsoftware/spine-core";
 
-export type SpineAssetBundle = {
+type SpineAssetPath = {
   skeletonPath: string;
-  skeletonExtension: string;
-  skeletonTextData?: string | ArrayBuffer;
   atlasPath: string;
   imagePaths: string[];
   imageExtensions: string[];
-}
+};
 
-type SpineLoaderParams =  {
+type SpineLoadContext = {
+  fileName: string;
+  atlasRefId?: string;
+  isSingleUrl: boolean;
+  spineAssetPath: SpineAssetPath;
+  skeletonRawData: string | ArrayBuffer;
+};
+
+type SpineLoaderParams = {
   fileExtensions?: string | string[];
-}
+};
 
 type SpineLoadItem = LoadItem & { params?: SpineLoaderParams };
 
 @resourceLoader("spine", ["json", "bin", "skel"])
 export class SpineLoader extends Loader<SpineResource> {
-  static imageExtensions = ["png", "jpg", "webp", "jpeg", "ktx", "ktx2"];
-  static skeletonExtensions = ["skel", "json", "bin"];
+  private static _imageExtensions = ["png", "jpg", "webp", "jpeg", "ktx", "ktx2"];
+  private static _skeletonExtensions = ["skel", "json", "bin"];
+  private static _BINARY_PREFIX = "spine:skel";
+  private static _decoder = new TextDecoder("utf-8");
 
-  static parseAndAssignSpineAsset(url: string, fileExtension: string | null, bundle: SpineAssetBundle) {
-    const { imageExtensions, skeletonExtensions } = SpineLoader;
-    const ext = SpineLoader.getUrlExtension(url, fileExtension);
+  /** @internal */
+  static _JSON_PREFIX = '{"spine":"json"';
+
+  /** @internal */
+  static _ATLAS_PREFIX = '{"data":"';
+
+  /** @internal */
+  static _isEditorBinary(reader: BufferReader): boolean {
```

Comment:
We need to discuss the value and necessity of asset specialization on spine！

I see huge maintenance costs

Reply:
done